inc/shortcodes/class-pdf.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Shortcake_Bakery\Shortcodes;
+
+class PDF extends Shortcode {
+
+	public static function get_shortcode_ui_args() {
+		return array(
+			'label'          => esc_html__( 'PDF', 'shortcake-bakery' ),
+			'listItemImage'  => 'dashicons-media-document',
+			'attrs'          => array(
+				array(
+					'label'  => esc_html__( 'URL', 'shortcake-bakery' ),
+					'attr'   => 'url',
+					'type'   => 'text',
+				),
+			),
+		);
+	}
+
+	/**
+	 * Turn embed code into a proper shortcode
+	 *
+	 * @param  string $content
+	 * @return string $content
+	 */
+	public static function reversal( $content ) {
+		$needle = '#http[^<]//[^<]+\.pdf?#';
```

Comment:
> But other than fixing the regex, it looks good to me...

And, once we figure out the appropriate regex for this one, we get #5 and everything else.

@davisshaver You might want to just copy the regex core uses for matching oEmbed providers.